### PR TITLE
feat(headless): Add Insight QuerySummary

### PR DESCRIPTION
**Pull request by aelfaleh**
https://coveord.atlassian.net/browse/SVCC-2313

### CODE REVIEW

**Review by lbergeron — APPROVED** [review]
Seems correct to me.

File: packages/headless/src/controllers/core/query-summary/headless-core-query-summary.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const searchStatus = buildCoreStatus(engine);
+  const getState = () => engine.state;
+
+  const durationInSeconds = () => {
+    const inSeconds = getState().search.duration / 1000;
+    return Math.round((inSeconds + Number.EPSILON) * 100) / 100;
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/headless/src/controllers/core/query-summary/headless-core-query-summary.ts:81]: Not entirely sure why, but it was there before https://github.com/coveo/ui-kit/commit/1df2414f4b2e87600eb44613c4cbe18d1c86add3
cc @olamothe

**Review by ThibodeauJF — APPROVED** [review]

File: packages/headless/src/controllers/core/query-summary/headless-core-query-summary.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lbergeron) [reply packages/headless/src/controllers/core/query-summary/headless-core-query-summary.ts:81]: The only reason I can think of would be to ensure we're dividing a non-zero number by 100. But, as far as I can see, `0 / 100` returns `0`, so it should not be an issue.

Anyways, having the `Number.EPSILON` there does not cause any harm, I am just curious.

**Review by nathanlb — APPROVED** [review]
Why are there changes to the package-lock files? 🤔

**Pull request merged**